scripts/core/get-superdesk-api-implementation.tsx

```diff
@@ -115,8 +123,12 @@ export function getSuperdeskApiImplementation(
     return {
         dataApi: dataApi,
         dataApiByEntity,
+        elasticsearch: elasticsearchApi,
```

Comment:
It would be very useful to remove the knowledge about elastic from the client and instead have an API with a generic query format for all endpoints that could work with any back-end whether it is mongo, elastic or if we add anything else.

I've created SDESK-4872 to address this after a discussion with Petr last December #3243.

Most of the time our queries are quite straight-forward and contain mostly of `=`, `<`, `>` and `contains` operators so it shouldn't be too hard to implement this query format.

We have to start somewhere, so I suggest first doing it for SAMS and then we could add it to other endpoints. I'd use a subset of mongo query format starting with [comparison](https://docs.mongodb.com/manual/reference/operator/query-comparison/) + AND/OR operators. That should probably do it for most part.

What do you think @MarkLark86 @petrjasek?

Reply:
This will definitely be a good improvement to add to the SAMS API itself, but at a later date. I do see the benefits in having a storage agnostic search query language, but we have time constraints at the moment.